src/components/gas/GasSpeedButton.js

```diff
@@ -147,6 +149,12 @@ const GasSpeedButton = ({
   // (and leave the number only!)
   // which gets added later in the formatGasPrice function
   const price = useMemo(() => {
+    if (
+      txNetwork !== currentNetwork ||
+      asset?.uniqueId !== transactionIdentifier
```

Comment:
changes make sense to me, the only comment i have is the name `transactionIdentifier`. While I was looking at the code I was thinking where this identifier came from, even tho you explained in the PR description that it was the asset id

Reply:
what would you suggest? I was also thinking about the better name

Reply:
I would at least use either `assetUniqueId` or `transactionUniqueId` to correspond with the rest of the naming in code.